python/pyspark/ml/util.py

```diff
@@ -760,3 +762,127 @@ def _get_active_session(is_remote: bool) -> SparkSession:
+        """
+        if file_path != "":
+            with open(file_path, "wb") as f:
+                cloudpickle.dump((train_fn, args, kwargs), f)
+                return f.name
+
+        if save_dir == "":
+            save_dir = os.getcwd()
+
+        with tempfile.NamedTemporaryFile(dir=save_dir, delete=False) as f:
+            cloudpickle.dump((train_fn, args, kwargs), f)
+            return f.name
+
+    @staticmethod
+    def create_training_script_from_func(fn_output_save_path: str, training_script_save_path: str, pickled_fn_path: str, prefix_code:str ="", suffix_code:str = "") -> str:
+        """
+            Given a file containing a pickled function and arguments, this function will create a pytorch file
+            that will execute the function and pickle the functions outputs.
+
+            Parameters
+            ----------
+            fn_output_save_path: str
+                This is the location where the created file will save the pickled output of the function.
+
+            training_script_save_path: str
+                This is the path which will be used for the created pytorch file.
+
+            pickled_fn_path:
+                This is the path of the file containing the pickled function, args, and kwargs.
+            
+            prefix_code: str
+                This contains a string that the user can pass in which will be executed before
+                the code generated by this class to execute the function and save it. If 
+                prefix_code is the empty string, nothing will be written before the auto-
+                generated code.
+
+            suffix_code: str
+                This contains a string of code that the user can pass in which will be executed
+                after the code generated by this class finishes executing. If suffix_code is 
+                the empty string, nothing will be written after the auto-generated code.
+            
+            Returns
+            -------
+            str:
+                The path to the location of the newly created pytorch file.
+        """
+
+        code_snippet =  textwrap.dedent(
+            f"""
+                    from pyspark import cloudpickle
+                    import os
+
+                    if __name__ == "__main__":
+                        with open("{pickled_fn_path}", "rb") as f:
+                            train_fn, args, kwargs = cloudpickle.load(f)
+                        output = train_fn(*args, **kwargs)
+                        with open("{fn_output_save_path}", "wb") as f:
+                            cloudpickle.dump(output, f)
+                    """
+        )
+        with open(training_script_save_path, "w") as f:
+            if prefix_code != "":
+                f.write(prefix_code)
+            f.write(code_snippet)
+        
+        return training_script_save_path
+    
+    
+    @staticmethod
+    def get_pickled_output_from_training_func(func_output_path: str) -> Any:
```

Comment:
Rename this to get_fn_output (output is not pickled)

Reply:
Done.